### PR TITLE
[Backport 2.x] Prevent message collection from being updated after message count has been received (#2180)

### PR DESCRIPTION
**Pull request by DarshitChanpura**
Backports ba9d82ef6a2c6da137c19fcaaddce6dabcf7160f from #2180 

Manual backport is required because of:
1. Conflicts between HTTP5 vs HTTP4  in the files in main vs 2.x.
2. Some code refactoring in TestAuditLogImpl file

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
Build is broken due to https://github.com/opensearch-project/OpenSearch/pull/8782

**Comment by DarshitChanpura** [comment]
Ci should be fixed once #3036 is merged.

File: src/main/java/org/opensearch/security/auditlog/impl/AuditMessage.java [hunk 1]
```diff
@@ -451,15 +451,27 @@ public String getExceptionStackTrace() {
...
+    }
+
+	@Override
+	public String toString() {
+		try {
+            return org.opensearch.common.Strings.toString(JsonXContent.contentBuilder().map(getAsMap()));
+		} catch (final IOException e) {
+		    throw ExceptionsHelper.convertToOpenSearchException(e);
+		}
+	}
```
Comment (DarshitChanpura) [inline src/main/java/org/opensearch/security/auditlog/impl/AuditMessage.java:473]: Looks weirdly formatted here, but that is not the case.

**Comment by cwperks** [comment]
@DarshitChanpura FYI the assertion for the failing test was changed on main to only check for 1 message: https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/auditlog/integration/BasicAuditlogTest.java#L145C47-L145C47

The test can be updated in this backport with the same assertion.

**Comment by cwperks** [comment]
@DarshitChanpura I see why the assertion on main was set to `1`. On main the RestHelper has automatic retries disabled. See https://github.com/opensearch-project/security/pull/2367/files#diff-8a5f07d4e22f127c17298eeebe6ea7bb6b49dd7b2ebd14398b7298627d8ed6d7R401

I think we can disable automatic retries on 2.x and set the assertion in BasicAuditlogTests.testSSLPlainText to check for 1 message

**Review by cwperks — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]

**Pull request merged**